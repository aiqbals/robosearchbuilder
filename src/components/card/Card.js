import React from 'react'; // to process JSX element
import { shallow, mount, render } from 'enzyme';

const Card = (props) => {
    const {id, name, email} = props; //destruturing
    //const Card = ({id, name, email}) => {} //destruturing, it can also be added as func parametter

    return(
        //<h1> Robo search building </h1>
        <div className=' tc bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5'>
            <img src={`https://robohash.org/${id}?200x200`} alt="robot"/>
            <div>
                <h3>{name}</h3>
                <h3>{email}</h3>
            </div>
        </div>
    );
}

export default Card;