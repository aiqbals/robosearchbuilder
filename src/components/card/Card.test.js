import React from 'react';
import { shallow, mount, render } from 'enzyme';
import Card from './Card';

it('expect to render Card component alone', () => {
    //expect(shallow(<Card />).length).toEqual(1) // enzyme test
    expect(shallow(<Card />).length).toMatchSnapshot(); // snapshot test
})

//console.log(shallow(<Card />))