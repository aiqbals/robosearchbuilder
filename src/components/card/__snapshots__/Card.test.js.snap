// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Card component alone 1`] = `1`;
